### ISSUE TITLE
feature: Update `:messages` on the fly when newlines are added

### ISSUE THREAD

**Issue by karma-riuk** [folke/noice.nvim#544]
### Did you check the docs?

- [X] I have read all the noice.nvim docs

### Is your feature request related to a problem? Please describe.

Basically, commands that take a long time to complete freeze the cmdline until they are done and the ouptut get dumped only once the command is finished. A concrete use-case of this problem is when using [`vim-fugitive`](https://github.com/tpope/vim-fugitive) with the `:Git push` command. In the example I show below however, I use the simple bash command to illustrate what I mean, but the idea is the same: updating the `messages` buffer as new lines of output are given.

### Describe the solution you'd like

I would like the way normal (neo)vim handles messages, where each line that gets outputted by the command gets added to the `messages` buffer so that we can notice the progress of the command, instead of waiting for it to be completed and having to parse through all the lines to see if it was a success or not.

### Describe alternatives you've considered

The alternatives I have considered was to simply disabling the messages of `noice`, but I don't think it is optimal. I like the idea of having a buffer to get the output of a command as it is much easier to scroll through, to search, manipulate, etc. But the feature of seeing each line shown when it actually happens is pretty important to me, as I am dyslexic and reading a huge block of text that appears at once is not as easy for me as single lines spread over time.

### Additional context

Here is an example with the command 
```bash
sleep 1; echo 1; sleep 2; echo 2
```
As mentionned previously, I first noticed the 'issue' when performing the `:Git push` command, but the idea is the same and this command is more easily reproducable.

https://github.com/folke/noice.nvim/assets/30158492/1ed8b962-65bb-4b59-a2dc-15511239cbc6

In the video 
1. I show the result of the command with the how `noice` works currently (notice the amount of time that passes between the moment I press '<CR>' and the output getting displayed)
2. I disable the `messages` functionality of `noice`
3. I repeat the command to show what normal neovim does with the output (the delay between the launch the command and the first line of output is greatly reduced as opposed to point 1.).

I hope this is not a duplicate of a pre-existing issue. I have tried to search for it but I struggled to find relevant keywords to describe the matter at hand. I apologise in advance if this subject was already brought and (maybe) solved.

**Comment by folke** [comment]
That's unfortunately not possible due to how Neovim's `ui_attach` works

**Comment by karma-riuk** [comment]
I understand, but would it then be possible to add a view that is the old neovim `:messages` instead of nui's `split`?

**Comment by folke** [comment]
Not possible. It's all or nothing
